File: client/src/models/Restaurant.js
import { Profile } from "./Account.js"



export class Restaurant{
  constructor(data){
    this.id = data.id
    this.name = data.name
    this.coverImg = data.coverImg
    this.description = data.description
    this.ownerId = data.ownerId
    this.owner = new Profile(data.owner)
    this.visitCount = data.visitCount
    this.reportCount = data.reportCount
    this.shutdown = data.shutdown
  }
}

File: server/Models/Restaurant.cs


public class Restaurant : RepoItem<int>
{
  public string Name { get; set; }
  public string CoverImg { get; set; }
  public string Description { get; set; }
  public string OwnerId { get; set; }

  public int VisitCount { get; set; }
  public int ReportCount { get; set; } // we will come back to this one
  public bool Shutdown { get; set; }
  public Profile Owner { get; set; }
}